internal/docker/docker_test.go

```diff
@@ -438,3 +551,213 @@ func TestToGenerateRequestJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestReadResponseJson(t *testing.T) {
+	t.Parallel()
+	contentLoader := func(data []byte, state *config.LibraryState) error {
+		return json.Unmarshal(data, state)
+	}
+	for _, test := range []struct {
+		name         string
+		jsonFilePath string
+		wantState    *config.LibraryState
+	}{
+		{
+			name:         "successful-unmarshal",
+			jsonFilePath: "../../testdata/successful-unmarshal-libraryState.json",
+			wantState: &config.LibraryState{
+				ID:                  "google-cloud-go",
+				Version:             "1.0.0",
+				LastGeneratedCommit: "abcd123",
+				APIs: []*config.API{
+					{
+						Path:          "google/cloud/compute/v1",
+						ServiceConfig: "example_service_config.yaml",
+					},
+				},
+				SourcePaths:   []string{"src/example/path"},
+				PreserveRegex: []string{"example-preserve-regex"},
+				RemoveRegex:   []string{"example-remove-regex"},
+			},
+		},
+		{
+			name:         "empty libraryState",
+			jsonFilePath: "../../testdata/empty-libraryState.json",
+			wantState:    &config.LibraryState{},
+		},
+		{
+			name:      "invalid_file_name",
+			wantState: nil,
+		},
+		{
+			name:         "invalid content loader",
+			jsonFilePath: "../../testdata/invalid-contentLoader.json",
+			wantState:    nil,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			tempDir := t.TempDir()
+			if test.name == "invalid_file_name" {
+				filePath := filepath.Join(tempDir, "my\x00file.json")
+				_, err := readResponse(contentLoader, filePath)
+				if err == nil {
+					t.Errorf("readResponse() expected an error but got nil")
+				}
+
+				assert.Contains(t, err.Error(), "failed to read response JSON file")
```

Comment:
Rewrite all these asserts as ordinary Go code. You can avoid duplication like so:
```
if g, w := err.Error(), "failed to..."; !strings.Contains(g, w) {
   t.Errorf("got %q, wanted it to contain %q", g, w)
}
```

Reply:
Done.